src/v/storage/disk_log_impl.h

```diff
@@ -380,6 +381,7 @@ class disk_log_impl final : public log {
     std::optional<model::offset> _cloud_gc_offset;
     std::optional<model::offset> _last_compaction_window_start_offset;
     size_t _reclaimable_size_bytes{0};
+    bool _compaction_enabled;
```

Comment:
Does `_compaction_enabled` need to exist? Does it ever diverge from `config().is_compacted()`?

Reply:
Yes, this stores the current state of the log's compaction status and it's used to compare against `config().is_compacted()` for changes in `disk_log_impl::notify_compaction_update`. We can't just use `config().is_compacted()` because when the `log_cleanup_policy` cluster config changes and we try to determine whether we need to (un)mark the segments as compacted, we need a way to remember what the old compaction status was.